### PR TITLE
implemented JWT support

### PR DESCRIPTION
**Pull request by nshtykh**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -86,6 +91,26 @@
             <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
             <version>2.1.0</version>
         </dependency>
+
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-api</artifactId>
+            <version>0.11.5</version>
```
Comment (andrii-hoienko) [inline pom.xml]: Please use variables to define versions

File: src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't find user by email"));
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java]: Specify the input email in the error message

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]: Please don`t forget about constants

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]: magic number

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

File: src/main/resources/db/changelog/changes/03-create-roles-table.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+                    unique: true
+
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/03-create-roles-table.yaml]: Please add empty line in the end of file

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Please change the base branch for this PR, because there are changes from previous one.

**Review by bhdnchui — APPROVED** [review]
Good job!

**Pull request merged**